### PR TITLE
feat: pull pdp data from contentfull

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: lib/operations/get-page/index.ts [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
+  const response = await contentful.fetchContent()
+  return response
+}
+
+export const getPage = async (params: PageProps) => {
   const currentCMS = publicRuntimeConfig.cms || ''
-  if (currentCMS === 'contentstack') {
-    return getContentStackPage(params)
-  }
+  if (currentCMS === 'contentstack') return getContentStackPage(params)
```
Comment (kibo-chandradeeptalaha) [inline lib/operations/get-page/index.ts:28]: Let's put these string values in constant or create enums

File: lib/operations/get-page/index.ts [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
   const response = await Stack.getEntry(params)
   return {
     components: response[0][0]?.page_components || [],
   }
 }
 
-export const getPage = async (params: any) => {
-  const { publicRuntimeConfig } = getConfig()
+const getContentfulPage = async () => {
+  const response = await contentful.fetchContent()
```
Comment (kibo-chandradeeptalaha) [inline lib/operations/get-page/index.ts:22]: directly return instead of storing in variable first

File: lib/operations/get-page/index.ts (hunk 1 above)
Comment (amolpawar-kibo) [reply lib/operations/get-page/index.ts]: Resolved.

File: lib/operations/get-page/index.ts (hunk 2 above)
Comment (amolpawar-kibo) [reply lib/operations/get-page/index.ts]: Done.

**Review by TriptiVermaXP — APPROVED** [review]

**Review by TriptiVermaXP — APPROVED** [review]
Looks good, but do we need to write tests for these?

File: lib/getters/contentfulGetters.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const formatCMSProduct = (cmsProducts: any) => {
```
Comment (kibo-chandradeeptalaha) [inline lib/getters/contentfulGetters.ts:1]: change function name to something related to formatContentfulProduct or normalizeContentfulProduct.

File: lib/cms/contentful/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import getHomePageQuery from './home-page-query'
+import getProductDetails from './product-details-query'
+
+const { publicRuntimeConfig } = getConfig()
+
+const accessToken = publicRuntimeConfig.contentful.accessToken
+const spaceId = publicRuntimeConfig.contentful.spaceId
+
+const URL = `https://graphql.contentful.com/content/v1/spaces/${spaceId}/environments/master`
```
Comment (kibo-chandradeeptalaha) [inline lib/cms/contentful/index.js:12]: Can we make this configurable ?

**Review by kibo-chandradeeptalaha — CHANGES_REQUESTED** [review]
Please take a look

File: lib/getters/contentfulGetters.ts (hunk 3 above)
Comment (amolpawar-kibo) [reply lib/getters/contentfulGetters.ts]: I have modify with formatContentfulProduct but in future we have to decide on CMS response.

**Pull request merged**